src/Dfc.CourseDirectory.WebV2/Features/Venues/AddVenue/Address.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.Search;
+using Dfc.CourseDirectory.Core.Search.Models;
+using Dfc.CourseDirectory.Core.Validation;
+using Dfc.CourseDirectory.Core.Validation.VenueValidation;
+using FluentValidation;
+using FluentValidation.Results;
+using FormFlow;
+using MediatR;
+using OneOf;
+using OneOf.Types;
+
+namespace Dfc.CourseDirectory.WebV2.Features.Venues.AddVenue.Address
+{
+    public class Query : IRequest<Command>
+    {
+    }
+
+    public class Command : IRequest<OneOf<ModelWithErrors<Command>, Success>>
+    {
+        public string AddressLine1 { get; set; }
+        public string AddressLine2 { get; set; }
+        public string Town { get; set; }
+        public string County { get; set; }
+        public string Postcode { get; set; }
+    }
+
+    public class Handler :
+        IRequestHandler<Query, Command>,
+        IRequestHandler<Command, OneOf<ModelWithErrors<Command>, Success>>
+    {
+        private readonly JourneyInstance<AddVenueJourneyModel> _journeyInstance;
```

Comment:
As comments from Controller, not on a fan of declaring the instance at this level as it doesn't need to be here. Better suited to a scoped variable.